src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketsTelemetry.cs

```diff
@@ -75,28 +77,33 @@ private void AcceptFailed(SocketError error, string? exceptionMessage)
         }
 
         [NonEvent]
-        public void ConnectStart(Internals.SocketAddress address)
+        public static void ConnectStart(Internals.SocketAddress address)
         {
-            if (IsEnabled(EventLevel.Informational, EventKeywords.All))
+            Interlocked.Increment(ref _currentOutgoingConnectAttempts);
```

Comment:
Why pay for additional interlocked operations if tracing isn't enabled?  Yes, we'd have an incorrect number if tracing was started later.  Isn't that ok?

Reply:
We have to know whether we counted a given connect as started to decide whether to decrement the counter afterwards.
That means we either pass around a flag indicating whether telemetry was enabled for `Start` (something we do for HTTP requests) or just always count them regardless of telemetry being on/off.

> Why pay for additional interlocked operations

Given this is only for Connect/Accepts, do we care about the overhead of 2 Interlocked operations? I feel it is a good tradeoff compared to storing more state and doing checks in callers.

Reply:
> We have to know whether we counted a given connect as started to decide whether to decrement the counter afterwards.

Do we?  Is it really so bad if the number is off a little if you start tracing in the middle of an operation?

Reply:
If it meant that you just don't observe the connects that were already in flight as you turn on the counter, that's fine.

But say you have a steady-state where there are N connects in flight. If you then turn on counters, you would observe a steady-state of ~0 connects.

Reply:
> Did measuring show zero impact on benchmarks (in particular ones that create and tear down lots of connections)?

I'm not seeing a noticeable difference doing `Connection: close` on citrine machines (28 thread)

Reply:
Ok